### PR TITLE
Modify restart function to not use system or moveset

### PR DESCRIPTION
**Pull request by rsdefever**
This PR represents a major overhaul to the `mc.restart()` function. Following this PR, the function no longer requires a `System` and `MoveSet`. Instead, the function finds the input file from the original run and modifies it accordingly to create an input file for the restart.

* Additional unit tests still required.

### CODE REVIEW

File: mosdef_cassandra/runners/runners.py [hunk 1]
```diff
@@ -92,66 +95,80 @@ def run(system, moveset, run_type, run_length, temperature, **kwargs):
...
-    assumes they already exist. No new fragment libraries are generated,
-    again, these should already exist from the original run. The maximum
-    translation, rotation, and volume move sizes are read from the checkpoint
-    file so the values specified in mc.Moves are **not** used. Similarly,
-    the starting structure is taken from the checkpoint file. The
-    keyword argument "restart_name" should specify the old "run_name",
-    i.e., the "run_name" that you are restarting from. If the "restart_name"
-    is not provided or if the "run_name" is the same as "restart_name",
-    "-rst" will be appended to the "run_name".
+    If you wish to extend a simulation you will need to specify the _total_
```
Comment (rmatsum836) [inline mosdef_cassandra/runners/runners.py:114]: I like this clarification.

File: mosdef_cassandra/runners/runners.py [hunk 2]
```diff
@@ -92,66 +95,80 @@ def run(system, moveset, run_type, run_length, temperature, **kwargs):
...
-    moveset : mosdef_cassandra.MoveSet
-        the MoveSet to simulate
-    run_type : "equilibration" or "production"
+    restart_from: str, optional, default=None
+        name of run to restart from; if None, searches current
+        directory for Cassandra inp files
+    run_length: int, optional, default=None
+        total length of the MC simulation; if None, use original simulation length
+    run_name: str, optional, default=None
+        name of this run; if None, appends "_rst" to run_name
```
Comment (rmatsum836) [inline mosdef_cassandra/runners/runners.py:127]: Line 111 says that `.rst.` will be appended to a filename but 127 says `_rst` will be appended.  `_rst` is the string that is appended to a restart file name in #90.  I don't have a preference either way, but just want to be consistent.

Comment (rsdefever) [reply mosdef_cassandra/runners/runners.py:127]: Good catch. I have moved to `.rst.` so I can use a sequential numbering scheme, `.rst.001.`, `.rst.002.`, ...

**Pull request merged**